internal/pkg/db/redis/data.go

```diff
@@ -1089,11 +1089,21 @@ func addEvent(conn redis.Conn, e correlation.Event) (id string, err error) {
 }
 
 func deleteEvent(conn redis.Conn, id string) error {
+	o, err := eventByID(conn, id)
+	if err != nil {
+		if err == redis.ErrNil {
+			return db.ErrNotFound
+		}
+		return err
+	}
+
 	_ = conn.Send("MULTI")
 	_ = conn.Send("UNLINK", id)
 	_ = conn.Send("UNLINK", db.EventsCollection+":readings:"+id)
 	_ = conn.Send("ZREM", db.EventsCollection, id)
 	_ = conn.Send("ZREM", db.EventsCollection+":created", id)
+	_ = conn.Send("ZREM", db.EventsCollection+":pushed", id)
+	_ = conn.Send("ZREM", db.EventsCollection+":device:"+o.Device, id)
```

Comment:
please also delete checksum if o.Checksum != ""

Reply:
I'm not seeing any `checksum` fields added anywhere, can you provide steps to duplicate so I can verify I'll be removing the correct thing?

Reply:
I found what you were talking about, addressing.